frontend/src/features/whats-new/utils/utils.ts

```diff
@@ -4,28 +4,22 @@ import { UserDto } from '~shared/types/user'
 
 import { FeatureUpdateList } from '../FeatureUpdateList'
 
-export interface shouldShowLatestFeatureUpdateNotificationProps {
-  user: UserDto | undefined
-  isLoading: boolean
-}
+export const getShowLatestFeatureUpdateNotification = (
+  user: UserDto | undefined,
+): boolean => {
+  if (user?.flags?.lastSeenFeatureUpdateDate) {
+    const sortedFeatureUpdateList = FeatureUpdateList.sort(
+      (featureUpdateA, featureUpdateB) =>
+        Date.parse(featureUpdateB.date) - Date.parse(featureUpdateA.date),
+    )
+    const lastSeenUserFeatureUpdateDate = new Date(
+      user?.flags?.lastSeenFeatureUpdateDate,
+    )
 
-export const shouldShowLatestFeatureUpdateNotification = ({
-  user,
-  isLoading,
-}: shouldShowLatestFeatureUpdateNotificationProps): boolean => {
-  if (isLoading) {
-    return false
-  }
-  const sortedFeatureUpdateList = FeatureUpdateList.sort(
-    (featureUpdateA, featureUpdateB) =>
-      Date.parse(featureUpdateB.date) - Date.parse(featureUpdateA.date),
-  )
-  const lastSeenUserFeatureUpdateDate = new Date(
-    user?.flags?.lastSeenFeatureUpdateDate ??
-      sortedFeatureUpdateList[sortedFeatureUpdateList.length - 1].date,
-  )
+    const latestFeatureUpdateDate = new Date(sortedFeatureUpdateList[0].date)
 
-  const latestFeatureUpdateDate = new Date(sortedFeatureUpdateList[0].date)
+    return isAfter(latestFeatureUpdateDate, lastSeenUserFeatureUpdateDate)
+  }
 
-  return isAfter(latestFeatureUpdateDate, lastSeenUserFeatureUpdateDate)
+  return true
```

Comment:
nit: prefer early return, reverse conditional check?
should we also use more date-fns?
e.g.

```tsx
if (!user?.flags?.lastSeenFeatureUpdateDate) {
  return true
}

const { date: firstDate } = FEATURE_UPDATE_LIST
  .sort((a, b) => compareDesc(a.date, b.date))
  .shift()

return isAfter(firstDate, user.flags.lastSeenFeatureUpdateDate)
```

Reply:
That is very neat. I did not think of using date-fns for comparison since I was always so used to using the Javascript sort function. Thanks for this suggestion.